packages/cli/src/command_builder.ts

```diff
@@ -13,85 +13,126 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-import yargs from 'yargs'
-import { ParsedCliInput, CliOutput, CliCommand, SpinnerCreator } from './types'
-import { Filter } from './filter'
-
-export type CommandBuilder<
-  TArgs = {},
-  TParsedCliInput extends ParsedCliInput<TArgs> = ParsedCliInput<TArgs>,
-  > =
-  // Create a CliCommand given a parsed CLI input (output of yargs parser) and output interface
-  (input: TParsedCliInput, output: CliOutput, spinner: SpinnerCreator) => Promise<CliCommand>
-
-export interface KeyedOptions { [key: string]: yargs.Options }
-export interface PositionalOptions { [key: string]: yargs.PositionalOptions }
-
-export interface YargsModuleOpts {
-  // Name of this command in the CLI, e.g., 'deploy'
-  // If positional arguments are included, they also need to be specified here
-  // See: https://github.com/yargs/yargs/blob/master/docs/advanced.md#positional-arguments
-  command: string
-
-  // Additional or shorthand names, e.g, 'a'
-  aliases?: string[]
+import { types, collections } from '@salto-io/lowerdash'
+import { Telemetry, CommandConfig } from '@salto-io/core'
+import { CliOutput, SpinnerCreator, CliExitCode, CliTelemetry } from './types'
+import { VERBOSE_OPTION } from './commands/commons/options'
+import { getCliTelemetry } from './telemetry'
 
-  // Description to be shown in help
-  description: string
+const { makeArray } = collections.array
 
-  // Positional arguments
-  positional?: PositionalOptions
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export type CommandOrGroupDef = CommandsGroupDef | CommandDef<any>
+
+export interface CommandsGroupDef {
+  properties: BasicCommandProperties
+  subCommands: CommandOrGroupDef[]
+}
 
-  // Keyed arguments
-  keyed?: KeyedOptions
+export type ActionInput<T> = {
+  input: T
+  telemetry: Telemetry
+  config: CommandConfig
+  output: CliOutput
+  spinnerCreator?: SpinnerCreator
+  workingDir?: string
```

Comment:
working directory has a different meaning, this argument is being used as the workspace directory, not the current working directory.
```suggestion
  workspacePath?: string
```